src/generator.js

```diff
@@ -365,3 +365,27 @@ export const degenRefiner = <CustomType: string, CustomImport: string>(
     { types: [], imports: [refinerSymbol], hoists: [] },
   ]
 }
+
+export const degenMaybe = <CustomType: string, CustomImport: string>(
+  refinerType: MetaType<CustomType, CustomImport>,
+  refiner: DeserializerGenerator<CustomType, CustomImport>,
+): DeserializerGenerator<CustomType, CustomImport> => {
+  const [code, dependencies] = refiner
+
+  return [
+    () => {
+      return `
+        (x: mixed): ?${refinerType.name} | Error => {
```

Comment:
Could you use `typeHeader(refinerType)` here instead of just the type's name? It will support parameterized types recursively.

Reply:
Done, with test coverage to boot.